### DISCUSSION TITLE
Issue installing GRR on security onion

### DISCUSSION THREAD

**Discussion by uskwarrior** [Security-Onion-Solutions/securityonion#2856] [Unsupported Versions]
Hello Folks,

I have been trying to install GRR Server on Security onion 16.04 and have been experiencing issues. Firstly I tried installing the docker image using the instructions here, but when logging to the server using hostname I get a proxy error. I tried installing the GRR as a service using instructions here, but each time when trying the first step of logging into SQL with the localhost password to create a grr database, I get an error "ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)". I have tried blank password and even resetting the password but found that mysql service doesn't let me. 

I have been scratching my head all day and haven't been able to resolve the issue. I would really appreciate any help or feedback on this.

**Comment by nezvocowa** [comment]
For the old SecurityOnion 16.04 the mysql password is randomly generated and is found in `/etc/mysql/debian.cnf`.

Example usage of mysql with the generated password: `sudo mysql --defaults-file=/etc/mysql/debian.cnf [COMMAND] [ARGUMENT]`
Most of it is documented in their old docs [16.04 documentation](https://docs.securityonion.net/en/16.04/mysql.html)

If you use a docker, remeber to [add paths to files or directories](https://docs.docker.com/storage/) you want that specific docker to be able to use, otherwise it won't be available in that containers namespace.

In the future, for 16.04-related questions, please use: [https://docs.securityonion.net/en/16.04/mailing-lists.html?highlight=mailing#security-onion-mailing-list](https://docs.securityonion.net/en/16.04/mailing-lists.html?highlight=mailing#security-onion-mailing-list)

**Reply by uskwarrior** [reply]
Thank you for your response.

Firstly, I tried this on ubuntu 18.04 using steps in this article <https://kifarunix.com/install-grr-incident-response-framework-on-ubuntu-18-04/> which worked.  However, it's not an option for me as I am supposed to use either Ubuntu 20.04 or Security onion 16.04 (which runs on Ubuntu 16.04 platform).

I was able to find the password for mysql database on Security onion using the steps you provided. I tried the same steps on Security onion, but got the error "Errors were encountered while processing: grr-server E: Sub-process /usr/bin/dpkg returned an error code (1)" (Please see screenshot). I have tried this on ubuntu 20.04 and got the same error.
![image](https://user-images.githubusercontent.com/78109598/106988069-d0942300-673c-11eb-8dc9-dad514e45552.png)

I tried installing the grr docker image using the steps specified in this article  <https://github.com/weslambert/securityonion-grr> which works but when trying to login to the console, I get error 502 (please see screenshot). I guess there are some changes needed on the docker file but there are no instructions for that.

I would really appreciate if you could guide me.

**Reply by nezvocowa** [reply]
The installer is complaining about python OpenSSL, so i would try to update the pyopenssl and then reinstalling?
[https://stackoverflow.com/questions/43267157/python-attributeerror-module-object-has-no-attribute-ssl-st-init](https://stackoverflow.com/questions/43267157/python-attributeerror-module-object-has-no-attribute-ssl-st-init)

Is there a reason why  you don't install the latest grr-server? That version is over two years old. (I have never used grr)

But on a side note Ubuntu 16.x is EOL in almost a month (April 30, 2021) and by the looks of grr's [documentation](https://grr-doc.readthedocs.io/en/latest/installing-grr-server/overview.html) grr is built for Ubuntu 18. So when installing a package on another version that it is built for specifically, anything can happen 😨

**Comment by uskwarrior** [comment]
Attached are the logs
[grr verbose log.txt](https://github.com/Security-Onion-Solutions/securityonion/files/5930107/grr.verbose.log.txt)

**Reply by weslambert** [reply]
I don't intend on updating the installer for 16.04, since it is going to be EOL soon anyway. You might consider using [Velociraptor on Security Onion 2](https://github.com/weslambert/securityonion-velociraptor) (although I need to test more recently).  If you have any other questions regarding 16.04, please post them here: https://docs.securityonion.net/en/16.04/mailing-lists.html?highlight=mailing#security-onion-mailing-list

**Comment by uskwarrior** [comment]
Thanks to both of you for your response.. apparently GRR server isnt
supported on either Ubuntu 16 or 18. Got to know this after breaking my
head for 3 days. I guess SO 2
3 is the way forward. Thanks again !
